File: Assets/Game/Hoftix/Script/GenerateScripts/UIModules/LoadingPanelView.cs
﻿using ZJYFrameWork.UISerializable.UIInitView;

namespace ZJYFrameWork.UISerializable
{
    public class LoadingPanelView : UIViewInterface
    {
        public UnityEngine.UI.Slider leftSlider_Slider = null;
        public ZJYFrameWork.UISerializable.LoadUIController LoadingController = null;
        public UnityEngine.UI.Text progressNum_Text = null;
        public UnityEngine.UI.Slider rightSlider_Slider = null;


        public void Init(UIView _view)
        {
            leftSlider_Slider = _view.GetObjType<UnityEngine.UI.Slider>("leftSlider_Slider");
            rightSlider_Slider = _view.GetObjType<UnityEngine.UI.Slider>("rightSlider_Slider");
            progressNum_Text = _view.GetObjType<UnityEngine.UI.Text>("progressNum_Text");
            LoadingController = _view.GetObjType<ZJYFrameWork.UISerializable.LoadUIController>("LoadingController");
        }
    }
}

File: DemoMMOUnity/Lua/Game/GenerateScripts/UIModules/LoadingPanelView.lua
---@class LoadingPanelView
LoadingPanelView = class("LoadingPanelView")
function LoadingPanelView:Init(view)
	self._UIView = view:GetComponent("UIView")
	self.leftSlider_Slider = self._UIView:GetObjTypeStr("leftSlider_Slider") or CS.UnityEngine.UI.Slider
	self.rightSlider_Slider = self._UIView:GetObjTypeStr("rightSlider_Slider") or CS.UnityEngine.UI.Slider
	self.progressNum_Text = self._UIView:GetObjTypeStr("progressNum_Text") or CS.UnityEngine.UI.Text
	self.LoadingController = self._UIView:GetObjTypeStr("LoadingController") or CS.UnityEngine.Object
end

